lib/redlock/client.rb

```diff
@@ -57,6 +58,23 @@ def unlock(lock_info)
       @servers.each { |s| s.unlock(lock_info[:resource], lock_info[:value]) }
     end
 
+    # Locks a resource, executing the received block only after successfully acquiring the lock,
+    # and returning its return value as a result.
+    # Params:
+    # +resource+:: the resource (or key) string to be locked.
+    # +ttl+:: the time-to-live in ms for the lock.
+    # +block+:: block to be executed after successful lock acquisition.
+    def with_lock(resource, ttl,  &block)
+      raise "No block passed" unless block
```

Comment:
personally I like this better:

```ruby
fail 'No block passed' unless block_given?
[...]
yield
```

No need to capture the block as a parameter here.